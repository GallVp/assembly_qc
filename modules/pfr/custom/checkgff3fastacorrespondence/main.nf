process CUSTOM_CHECKGFF3FASTACORRESPONDENCE {
    tag "$meta.id"
    label 'process_single'

    conda "${moduleDir}/environment.yml"
    container "${ workflow.containerEngine == 'singularity' || workflow.containerEngine == 'apptainer' ?
        'https://depot.galaxyproject.org/singularity/samtools:1.18--h50ea8bc_1':
        'quay.io/biocontainers/samtools:1.18--h50ea8bc_1' }"

    input:
    tuple val(meta), path(gff3)
    path(fasta)

    output:
    tuple val(meta), path('*.success.log')  , emit: success_log , optional: true
    tuple val(meta), path('*.error.log')    , emit: error_log   , optional: true
    path "versions.yml"                     , emit: versions

    when:
    task.ext.when == null || task.ext.when

    shell:
    prefix = task.ext.prefix ?: "${meta.id}"
    template 'check_gff3_fasta_correspondence.sh'
}
