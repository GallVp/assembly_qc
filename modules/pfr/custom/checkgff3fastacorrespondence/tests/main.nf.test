nextflow_process {

    name "Test Process CUSTOM_CHECKGFF3FASTACORRESPONDENCE"
    script "../main.nf"
    process "CUSTOM_CHECKGFF3FASTACORRESPONDENCE"

    tag "modules"
    tag "modules_nfcore"
    tag "custom"
    tag "custom/checkgff3fastacorrespondence"

    test("sarscov2-fasta-gff3-success") {

        when {
            process {
                """
                input[0] = [
                    [ id:'test' ], // meta map
                    file(params.test_data['sarscov2']['genome']['genome_gff3'], checkIfExists: true)
                ]

                input[1] = [
                    file(params.test_data['sarscov2']['genome']['genome_fasta'], checkIfExists: true)
                ]
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(process.out).match() },
                { assert process.out.error_log == [] },
                { assert process.out.success_log != null },
                { assert path(process.out.success_log.get(0).get(1)).getText().contains("All tests passed...")},
            )
        }

    }

    test("sarscov2-gff3-homo_sapiens-fasta-error") {

        when {
            process {
                """
                input[0] = [
                    [ id:'test' ], // meta map
                    file(params.test_data['sarscov2']['genome']['genome_gff3'], checkIfExists: true)
                ]

                input[1] = [
                    file(params.test_data['homo_sapiens']['genome']['genome_fasta'], checkIfExists: true)
                ]
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(process.out).match() },
                { assert process.out.success_log == [] },
                { assert process.out.error_log != null },
                { assert path(process.out.error_log.get(0).get(1)).getText().contains("GFF3 file contains identifiers not present in FASTA")},
            )
        }

    }

}
