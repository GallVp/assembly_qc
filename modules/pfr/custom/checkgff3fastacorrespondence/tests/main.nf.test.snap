{
    "sarscov2-gff3-homo_sapiens-fasta-error": {
        "content": [
            {
                "0": [
                    
                ],
                "1": [
                    [
                        {
                            "id": "test"
                        },
                        "test.error.log:md5,8a119170625dc95fb2faa6843fad2c3f"
                    ]
                ],
                "2": [
                    "versions.yml:md5,c8e0bb60f7422aa6c15db35013620802"
                ],
                "error_log": [
                    [
                        {
                            "id": "test"
                        },
                        "test.error.log:md5,8a119170625dc95fb2faa6843fad2c3f"
                    ]
                ],
                "success_log": [
                    
                ],
                "versions": [
                    "versions.yml:md5,c8e0bb60f7422aa6c15db35013620802"
                ]
            }
        ],
        "timestamp": "2023-11-29T12:24:08.677505"
    },
    "sarscov2-fasta-gff3-success": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test"
                        },
                        "test.success.log:md5,5cad27984e6af4889f7dcf12264fe47b"
                    ]
                ],
                "1": [
                    
                ],
                "2": [
                    "versions.yml:md5,c8e0bb60f7422aa6c15db35013620802"
                ],
                "error_log": [
                    
                ],
                "success_log": [
                    [
                        {
                            "id": "test"
                        },
                        "test.success.log:md5,5cad27984e6af4889f7dcf12264fe47b"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,c8e0bb60f7422aa6c15db35013620802"
                ]
            }
        ],
        "timestamp": "2023-11-29T12:24:04.530428"
    }
}